jerry-core/module/module.c

```diff
@@ -0,0 +1,265 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "module.h"
+#include "jerryscript.h"
+#include "jcontext.h"
+
+#include "ecma-helpers.h"
+#include "ecma-function-object.h"
+#include "ecma-literal-storage.h"
+#include "ecma-lex-env.h"
+#include "ecma-gc.h"
+#include "vm.h"
+
+#ifndef CONFIG_DISABLE_ES2015_MODULE_SYSTEM
+
+/**
+ * Check if property is exported from the script.
+ * @returns the export name - if the given property is exported
+ *          NULL - otherwise
+ */
+static parser_module_names_t *
+parser_module_is_property_exported (ecma_string_t *property_name_p, /**< property name */
+                                    parser_module_node_t *export_node_p) /**< export node */
+{
+  parser_module_names_t *current_p = export_node_p->module_names_p;
+
+  if (current_p == NULL)
+  {
+    return NULL;
+  }
+
+  for (uint16_t i = 0; i < export_node_p->module_request_count; i++)
+  {
+    parser_module_names_t *next_p = current_p->next_p;
+    ecma_string_t *import_name_p = ecma_new_ecma_string_from_utf8 (current_p->local_name.value_p,
+                                                                   current_p->local_name.length);
+
+    bool found = ecma_compare_ecma_strings (import_name_p, property_name_p);
+
+    ecma_deref_ecma_string (import_name_p);
+
+    if (found)
+    {
+      return current_p;
+    }
+    current_p = next_p;
+  }
+
+  return NULL;
+} /* parser_module_is_property_exported */
+
+/**
+ * Compare property name with imports.
+ * @return the export name - if the exported property is imported
+ *         NULL - otherwise
+ */
+static parser_module_names_t *
+parser_module_compare_property_name_with_import (parser_module_node_t *module_node_p, /**< module node */
+                                                 parser_module_names_t *export_names_p) /**< export names */
+{
+  parser_module_names_t *current_p = module_node_p->module_names_p;
+
+  if (current_p == NULL)
+  {
+    return NULL;
+  }
+
+  for (uint16_t i = 0; i < module_node_p->module_request_count; i++)
+  {
+    parser_module_names_t *next_p = current_p->next_p;
+
+    if (current_p->local_name.length == export_names_p->import_name.length
+        && memcmp (export_names_p->import_name.value_p,
+                   current_p->local_name.value_p,
+                   current_p->local_name.length) == 0)
+    {
+      return current_p;
+    }
+
+    current_p = next_p;
+  }
+
+  return NULL;
+} /* parser_module_compare_property_name_with_import */
+
+/**
+ * Connect the imported script's propeties to the main script.
+ */
+static void
+module_connect_properties (ecma_object_t *scope_p) /** scope */
```

Comment:
`/** scope */` -> `/**< scope */`

Reply:
Also add an assert for that `scope_p` should be declarative lexical environment.